mozilla/foundation.mozilla.org
# 9151 - related articles for Products, Categories, and Editorial Content Index

# Description

Related PRs: #9216 (Used this PR for reference)
Related issues: #9151 

This PR introduces:
- Related Articles for Product Pages
- Related Articles for Product Categories
- Related Articles for the Editorial Content Index Page
- Updated factory file to make sure all these models have related articles in new DBs
- Updated tests

# Steps to test

1.) Pull this branch
2.) Run `inv new-db`
3.) Visit each page type and verify there are articles
4.) If desired, log into the CMS and update related articles to any of these pages 


# Checklist



**Tests**
( NOTE: There are tests added, however, the test` test_related_articles_on_multiple_categories` seems to be returning one item too many and causing a failure. Please see discussion below for details. Per request of @mtdenton we have added the `unittest.skip` flag along with some details to keep this work moving forward since the actual functionality is working as expected, and will create a new ticket to investigate!)

**Changes in Models:**

**Documentation:**



Screenshots:
---

Category Page:
<img width="572" alt="Screen Shot 2022-08-23 at 6 24 52 PM" src="https://user-images.githubusercontent.com/18314510/186297240-be3ce859-29e5-4b35-9b73-844d0cd3ed78.png">

Product Page:
<img width="580" alt="Screen Shot 2022-08-23 at 6 33 05 PM" src="https://user-images.githubusercontent.com/18314510/186298006-478c9ae8-c8b4-469d-b6b4-51fab49eb7c5.png">

Editorial Content Index Page:
<img width="594" alt="Screen Shot 2022-08-24 at 3 13 17 PM" src="https://user-images.githubusercontent.com/18314510/186533223-68dadd85-bb8e-4a64-b64b-d90851d92c45.png">
